### DISCUSSION TITLE
Suddenly my nvim doesn't open file directly via command

### DISCUSSION THREAD

**Discussion by djkeh** [neovim/neovim#29616] [Q&A]
I'm using macbook air m1, oh my zsh + neovim installed.
If I command vi with filename, it just doesn't simply open the file, instead nvim shows starting screen to select specific files in the current dir.
I tested:

```sh
nvim filename
vi filename
```

My aliases are set to:

```
❯ which nvim
/opt/homebrew/bin/nvim
❯ which vim
/usr/bin/vim
❯ which vi
vi: aliased to nvim
```

After I found this problem I just removed the alias for vim, so it opens spacevim, and following command works fine.

```sh
vim filename
```

my alias part in `.zshrc` looks like:

```sh
# Set personal aliases, overriding those provided by oh-my-zsh libs,
# plugins, and themes. Aliases can be placed here, though oh-my-zsh
# users are encouraged to define aliases within the ZSH_CUSTOM folder.
# For a full list of active aliases, run `alias`.
#
# Example aliases
# alias zshconfig="mate ~/.zshrc"
# alias ohmyzsh="mate ~/.oh-my-zsh"
#alias vim="nvim"
alias vi="nvim"
alias vimdiff="nvim -d"
alias cat="bat"
export EDITOR=/opt/homebrew/bin/nvim
```

I think it's my fault, I just don't know where to start.
How do I fix this to load a file directly like `vi filename`, `nvim filename` as it was?
Thx.

**Comment by knarrff** [comment]
Hi @djkeh 

It might be useful to start excluding things.

Try ```nvim --clean filename```. This should not use any of your configuration files. If this works, at least you limited the problem to those and could start commenting things to see what happens If it still does not work, at least you now know that your vim-config files should not be the problem.

Depending on the outcome of above, ```nvim -V[N]``` might be helpful, or trying the same with a different user account, or at least a different $HOME set, to possibly include/exclude other user-related or system/installation-related problems.

**Reply by djkeh** [reply]
Hi @knarrff 
I tried your first command and it works. I also tried the verbose mode, which tells me so much information that I couldn't find what exactly was happening. I guess there would be no user-related problems because this is my local laptop and I'm the only user with the only macbook account. I think vim-config might be the problem, is it `.zshrc` that you mean?

**Reply by knarrff** [reply]
If ```nvim --clean``` did work, then I suspect the problem to be in your configuration. This includes .vimrc, or also installed plugins.

For .vimrc: rename it for a moment and see if that also helps (don't use --clean here). If it does, rename it back and start commenting out blocks, and after each change see if this affects the outcome. Doing so, bit by bit, you should get closer to the line(s) that make a difference.

Plugins are the second possibility, and maybe even the more likely since I would expect vim to complain if you would have an error in .vimrc. Similarly to .vimrc, first disable all to see if the problem is there. ```--noplugin``` could be used to disable all (but use .vimrc). If this helps, see which you have installed: within vim, type ```:scriptnames``` and see if something stands out - such as something recently installed or changed.

**Reply by djkeh** [reply]
Thx to your advice, I feel like I'm narrowing down the cause of this problem. `vi --noplugin filename` works. In my case I'm using oh my zsh, so the configuration file name is `.zshrc`. `.vimrc` is not in my home directory. I looked into `.zshrc` and here's the part of the plugin setting:

```sh
# Which plugins would you like to load?
# Standard plugins can be found in $ZSH/plugins/
# Custom plugins may be added to $ZSH_CUSTOM/plugins/
# Example format: plugins=(rails git textmate ruby lighthouse)
# Add wisely, as too many plugins slow down shell startup.
plugins=(
  git
  zsh-syntax-highlighting
  zsh-autosuggestions
  fzf
  fasd
  asdf
)

. $(brew --prefix asdf)/libexec/asdf.sh
source $ZSH/oh-my-zsh.sh
```

I disabled every line except `source $ZSH/oh-my-zsh.sh` by commenting it out to prevent using any plugins, reloaded the configuration, but it didn't work. It still starts from the init page like this:

<img width="816" alt="image" src="https://github.com/user-attachments/assets/1a95ef71-900f-456e-9187-671a4e7b3e48">

I think the problem stays in the configuration file, but for me there's no line to be suspected. Here's my entire configuration files without comments:

```sh
# .zprofile
iterm2_print_user_vars() {
  CPU=$(uname -m)
  iterm2_set_user_var cpu $CPU
}

CPU=$(uname -m)
if [[ "$CPU" == "arm64" ]]; then
  eval "$(/opt/homebrew/bin/brew shellenv)"
else
  export PATH=/opt/homebrew/bin:$PATH
  eval "$(/usr/local/bin/brew shellenv)"
  export FZF_BASE=/opt/homebrew/bin/fzf
  export ASDF_DATA_DIR=$HOME/.asdf-x86
fi

alias ibrew="arch -x86_64 /usr/local/bin/brew"
alias abrew="arch -arm64 /opt/homebrew/bin/brew"

export PATH="$PATH:/usr/local/bin"
```

```sh
# .zshrc
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi

export ZSH="$HOME/.oh-my-zsh"
ZSH_THEME="powerlevel10k/powerlevel10k"

plugins=(
  git
  zsh-syntax-highlighting
  zsh-autosuggestions
  fzf
  fasd
  asdf
)

. $(brew --prefix asdf)/libexec/asdf.sh
source $ZSH/oh-my-zsh.sh

#plugins=(
#  git
#  zsh-syntax-highlighting
#  zsh-autosuggestions
#  fzf
#  fasd
#  asdf
#)

#. $(brew --prefix asdf)/libexec/asdf.sh
source $ZSH/oh-my-zsh.sh

# To customize prompt, run `p10k configure` or edit ~/.p10k.zsh.
[[ ! -f ~/.p10k.zsh ]] || source ~/.p10k.zsh

# My custom settings
test -e "${HOME}/.iterm2_shell_integration.zsh" && source "${HOME}/.iterm2_shell_integration.zsh"
[[ -d ~/.rbenv  ]] && \
  export PATH=${HOME}/.rbenv/bin:${PATH} && \
  eval "$(rbenv init -)"

export PATH=~/bin:${PATH}
export PATH=${PATH}:$(brew --prefix python)/libexec/bin
export GOOGLE_APPLICATION_CREDENTIALS="/Users/uno/.ssh/GOOGLE_APPLICATION_CREDENTIALS.json"
```

Do you see anything suspicious?

**Reply by knarrff** [reply]
What I meant with "plugins" is vim plugins. But thinking again about it: in theory your shell could play a part in that as well. In order to exclude this, when you tried to open a file using vim and have vim still open (but on the init page), see which command was used to start vim, e.g., by using another shell: ```ps afxuw | grep vi```. This should show at least one line containing the command used to start vi with your filename. If the filename is there, I think the shell might not be the problem and it is time to look at vim plugins.

In order to see what you've got installed, type within vim: ```:scriptnames```.

**Reply by djkeh** [reply]
Oh. I misunderstood. I looked into the vim plugins again using `:scriptnames`, and checked there are 194 plugins loaded. None of them are installed by me, I think those were included in the oh my zsh installation process. If I can't find exactly which plugin caused the problem, I think I have no choice but to re-install oh my zsh.
